tcpchecker.go

```diff
@@ -0,0 +1,137 @@
+package checkup
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+)
+
+// TCPChecker implements a Checker for TCP endpoints.
+type TCPChecker struct {
+	// Name is the name of the endpoint.
+	Name string `json:"endpoint_name"`
+
+	// URL is the URL of the endpoint.
+	URL string `json:"endpoint_url"`
+
+	// TLSEnabled controls whether to enable TLS or not.
+	// If set, TLS is enabled.
+	TLSEnabled bool `json:"tls,omitempty"`
+
+	// TLSSkipVerify controls whether to skip server TLS
+	// certificat validation or not.
+	TLSSkipVerify bool `json:"tls_skip_verify,omitempty"`
+
+	// TLSCAfile is the Certificate Authority used
+	// to validate the server TLS certificate.
+	TLSCAfile string `json:"tls_cafile,omitempty"`
```

Comment:
Oh and "TLSCAFile" might technically be the correct casing, although I'm bikeshedding now.